### ISSUE TITLE
No way to enable plugin after it was disabled

### ISSUE THREAD

**Issue by OlegYch** [sbt/sbt#1926]
i have a bunch of projects and use sbt-assembly
i want sbt-assembly to be disabled by default (not sure why this is not the default)
so i use myProject(id, base) = Project(id, base).disablePlugins(Assembly).bunchofotherdefaults
in one particular project i want to enable the plugin, but i get an exception if i use enablePlugin

```
sbt.AutoPluginException: Error determining plugins for project 'keen-storefront' in D:\keen\storefront:
Contradiction in selected plugins.  These plugins were both included and excluded: sbtassembly.AssemblyPlugin
        at sbt.AutoPluginException.withPrefix(Plugins.scala:106)
```

imo the last setting should win just like it behaves everywhere, and there should be no exceptional cases

**Comment by jsuereth** [comment]
Yeah, this is a known issue in our logic system (and the translation of enablement/disablement into it).  I'm working on a mechanism to fix it.

PRs welcome if you have time.

**Comment by jvican** [comment]
There was recently a SO question about this: https://stackoverflow.com/questions/46779736/sbt-autoplugins-conflict-that-shouldnt-be-there/46835647#46835647.

**Comment by laughedelic** [comment]
I discovered this when I was asked by a user of [sbt-s3-resolver](https://github.com/ohnosequences/sbt-s3-resolver) who had a complicated multi-project build. He wanted to enable plugin in some subprojects, but not others (the plugin enables itself automatically). I think it's quite a common situation.

**Comment by myyk** [comment]
@laughedelic That was my situation years ago when I started following this issue.

**Comment by it-education-md** [comment]
the root cause is that we were accumulating plugin toggles as a raw conjunction, so `disablePlugins(X).enablePlugins(X)` left both `!X` and `X` in the selection and blew up with the contradiction error at load time. I'd like to make explicit toggles override the opposite prior toggle for the same plugin (last call wins).

if this direction sounds good, I can open a PR. cc @eed3si9n @xuwei-k

**Comment by eed3si9n** [comment]
@it-education-md Sure.
